libs/remote_storage/src/s3_bucket.rs

```diff
@@ -278,33 +307,57 @@ impl S3Bucket {
 
     async fn delete_oids(
```

Comment:
I think having both `delete_oids` and `delete_oids0` is redundant. The `delete_oids` function is only called once. The content of `delete_oids` should IMO be moved back into `delete_objects`, and `delete_oids0` should be renamed to `delete_oids`.

Reply:
I think I needed to do some permit wrestling, seeing about this soon.

Reply:
Ok, cleaned up this in [a2eeb13](https://github.com/neondatabase/neon/pull/6697/commits/a2eeb13bf55e3f3cb3b8d6b46523f6473858bf56)